### PR TITLE
PLANNER-1021: thread count hard coded fix

### PR DESCRIPTION
**Pull request by jgoldsmith613**

### CODE REVIEW

File: kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/OptaplannerKieServerExtension.java [hunk 1]
```diff
@@ -82,17 +82,16 @@ public void init(KieServerImpl kieServer, KieServerRegistry registry) {
...
+        int resolvedActiveThreadCount = availableProcessorCount <= 2 ? 1 : availableProcessorCount - 2;
+    
+       
         this.threadPool = new ThreadPoolExecutor(
-                Math.min(2, poolSize), // core size
-                poolSize, // max size
-                120, // idle timeout
-                                                 TimeUnit.SECONDS,
-                                                 new ArrayBlockingQueue<Runnable>(poolSize)); // queue with a size
+        		resolvedActiveThreadCount,
```
Comment (ron-murhammer) [inline kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/OptaplannerKieServerExtension.java]: Can you verify this is formatted properly? Looks like an extra tab here.

File: kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/OptaplannerKieServerExtension.java [hunk 2]
```diff
@@ -82,17 +82,16 @@ public void init(KieServerImpl kieServer, KieServerRegistry registry) {
...
         // if new jobs are submited and all threads are busy, the reject policy will kick in.
-        int poolSize = Runtime.getRuntime().availableProcessors();
-        if ( poolSize >= 4 ) {
-            // Leave 1 processor alone to handle REST/JMS requests and run the OS
-            poolSize--;
-        }
+        int availableProcessorCount = Runtime.getRuntime().availableProcessors();     
+        int resolvedActiveThreadCount = availableProcessorCount <= 2 ? 1 : availableProcessorCount - 2;
+    
+
```
Comment (ron-murhammer) [inline kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/OptaplannerKieServerExtension.java]: Probably don't need 2 empty lines here.

File: kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/OptaplannerKieServerExtension.java [hunk 3]
```diff
@@ -82,17 +82,16 @@ public void init(KieServerImpl kieServer, KieServerRegistry registry) {
         this.registry = registry;
         // the following threadpool will have a max thread count equal to the number of cores on the machine.
         // if new jobs are submited and all threads are busy, the reject policy will kick in.
-        int poolSize = Runtime.getRuntime().availableProcessors();
-        if ( poolSize >= 4 ) {
-            // Leave 1 processor alone to handle REST/JMS requests and run the OS
-            poolSize--;
-        }
+        int availableProcessorCount = Runtime.getRuntime().availableProcessors();     
+        int resolvedActiveThreadCount = availableProcessorCount <= 2 ? 1 : availableProcessorCount - 2;
```
Comment (ron-murhammer) [inline kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/OptaplannerKieServerExtension.java]: Could instead write this as `Math.max(1, availableProcessorCount - 2)`. That is probably a bit easier to read.

**Comment by triceo** [comment]
ok to test

File: kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/OptaplannerKieServerExtension.java (hunk 3 above)
Comment (ge0ffrey) [reply kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/OptaplannerKieServerExtension.java]: Doing this change in the optaplanner-core code that does something similar too.

**Comment by ge0ffrey** [comment]
@yurloc @jgoldsmith613 I've merged it. I do agree with Ron's suggestions, but this PR is already a step forward, so it was good to merge. jiri, if you can also fix Ron's suggestion in another PR (doesn't need to be a seperate one, but it can be), that would be nice.

File: kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/OptaplannerKieServerExtension.java (hunk 3 above)
Comment (ge0ffrey) [reply kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/OptaplannerKieServerExtension.java]: When it comes to soft things (such as code style, etc), we are more lenient with contributors than we are to full-time committers who have the time to make sure all the i's are dotted and all t's are crossed. We'll take care of dotting and crossing those.
Thanks for the contribution :)

**Pull request merged**